src/scores/continuous/consistent_impl.py

```diff
@@ -0,0 +1,262 @@
+        reduce_dims: Optionally specify which dimensions to reduce when
+            calculating the consistent Huber score. All other dimensions will be preserved. As a
+            special case, 'all' will allow all dimensions to be reduced. Only one
+            of `reduce_dims` and `preserve_dims` can be supplied. The default behaviour
+            if neither are supplied is to reduce all dims.
+        preserve_dims: Optionally specify which dimensions to preserve when calculating
+            the consistent Huber score. All other dimensions will be reduced. As a special case, 'all'
+            will allow all dimensions to be preserved. In this case, the result will be in
+            the same shape/dimensionality as the forecast, and the errors will be the consistent Huber
+            score at each point (i.e. single-value comparison against observed), and the
+            forecast and observed dimensions must match precisely. Only one of `reduce_dims`
+            and `preserve_dims` can be supplied. The default behaviour if neither are supplied
+            is to reduce all dims.
+        weights: Optionally provide an array for weighted averaging (e.g. by area, by latitude,
+            by population, custom)
+
+    Returns:
+        array of (mean) scores that is consistent for Huber mean functional,
+        with the dimensions specified by `dims`. If `dims` is `None`, the returned DataArray will have
+        only one entry, the overall mean score.
+
+    Raises:
+       ValueError: if `huber_param <= 0`.
+
+    References:
+        -   Taggart, R. (2021). Evaluation of point forecasts for extreme events using
+            consistent scoring functions. Quarterly Journal of the Royal Meteorological
+            Society, 148(742), 306–320. https://doi.org/10.1002/qj.4206
+        -   Taggart, R. J. (2022). Point forecasting and forecast evaluation with
+            generalized Huber loss. Electronic Journal of Statistics, 16(1).
+            https://doi.org/10.1214/21-ejs1957
+    """
+    check_huber_param(huber_param)
+    if preserve_dims or reduce_dims:
+        reduce_dims = gather_dimensions(fcst.dims, obs.dims, reduce_dims=reduce_dims, preserve_dims=preserve_dims)
+
+    kappa = (fcst - obs).clip(min=-huber_param, max=huber_param)
+    result = 0.5 * (phi(obs) - phi(kappa + obs) + kappa * phi_prime(fcst))
+    result = apply_weights(result, weights=weights)
+    result = result.mean(dim=reduce_dims)
+
+    return result
+
+
+def consistent_quantile_score(
+    fcst: xr.DataArray,
+    obs: xr.DataArray,
+    *,  # Force keywords arguments to be keyword-only
+    alpha: float,
+    g: Callable[[xr.DataArray], xr.DataArray],
+    reduce_dims: Optional[FlexibleDimensionTypes] = None,
+    preserve_dims: Optional[FlexibleDimensionTypes] = None,
+    weights: Optional[xr.DataArray] = None,
+) -> xr.DataArray:
+    """
+    Score that is consistent for the alpha-quantile functional, based on nondecreasing function g.
```

Comment:
It might be my programmer brain, but every time I read this, I image that one might be able to 'plug and play' and choose any number of given scores which meet the relevant consistency criteria, but I don't think that's what meant. I think what's happening is that the g function is essentially weighting or skewing the forecast and obs, and alpha is splitting the under and over forecast components at a cutoff. But maybe g is actually the score in disguise, and by supplying different "gs", the driver function might apply a mapping which is conceptually like a square or a square root function introducing different nonlinearities into the response of the function to the error. I realise I'm not a domain expert here, but it just keeps showing me.

Reply:
I think with (a) the link to the tutorials being included and (b) the main use likely being in threshold-weighted scores, I'm okay with each docstring not labouring the point. I eventually arrived at the proper understanding but it wasn't really clear just from the docstring where to start.